cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -157,6 +163,177 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
+            {
+                TypeBroadCastImpl<uint64_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<float>(arr_in))
+            {
+                TypeBroadCastImpl<float>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<double>(arr_in))
+            {
+                TypeBroadCastImpl<double>::broadcast(arr_out, arr_in);
+            }
+            else
+            {
+                throw std::runtime_error("input array data type not support!");
+            }
+        }
+
+        static void throw_shape_error(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            const auto & left_shape = arr_out.shape();
+            const auto * right_shape = arr_in.shape();
+
+            std::ostringstream msg;
+            msg << "Broadcast input array from shape(";
+            for (pybind11::ssize_t i = 0; i < arr_in.ndim(); ++i)
+            {
+                msg << right_shape[i];
+                if (i != arr_in.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ") into shape(";
+            for (size_t i = 0; i < arr_out.ndim(); ++i)
+            {
+                msg << left_shape[i];
+                if (i != arr_out.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ")";
+
+            throw std::runtime_error(msg.str());
+        };
+    };
+
+    template <typename D /* for destination type */>
+    struct TypeBroadCastImpl
+    {
+        static void broadcast(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            using out_type = typename std::remove_reference<decltype(arr_out[0])>::type;
+
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            auto * arr_new = reinterpret_cast<pybind11::array_t<D> *>(&arr_in);
+
+            shape_type sidx_init(arr_out.ndim());
+
+            for (size_t i = 0; i < arr_out.ndim(); ++i)
+            {
+                sidx_init[i] = 0;
+            }
+
+            std::function<void(shape_type, int)> copy_idx;
+            copy_idx = [&](shape_type sidx, int dim)
+            {
+                if (dim < 0)
+                {
+                    return;
+                }
+
+                for (size_t i = 0; i < arr_out.shape(dim); ++i)
+                {
+                    sidx[dim] = i;
+
+                    size_t offset_in = 0;
+                    for (pybind11::ssize_t it = 0; it < arr_in.ndim(); ++it)
+                    {
+                        offset_in += arr_in.strides(it) / arr_in.itemsize() * sidx[it];
+                    }
+
+                    const D * ptr_in = arr_new->data() + offset_in;
+                    // NOLINTNEXTLINE(bugprone-signed-char-misuse, cert-str34-c)
+                    arr_out.at(sidx) = static_cast<out_type>(*ptr_in);
+                    copy_idx(sidx, dim - 1);
```

Comment:
It's good to use a comment to signify the recursion.

Reply:
done